mirrormanager2/checkin.py

```diff
@@ -40,6 +41,34 @@
 logger = logging.getLogger(__name__)
 
 
+@APP.route('/checkin/json', methods=['POST'])
+def checkin_json():
+    ctype = flask.request.headers.get('Content-Type')
+    try:
+        if ctype == 'application/gzip':
+            data = json.loads(zlib.decompress(flask.request.data, 16+zlib.MAX_WBITS))
+        elif ctype == 'application/json':
+            data = json.loads(flask.request.data)
+        else:
+            return 'Invalid content type', 415
+    except Exception as ex:
+        logging.error('Unable to read input data: %s' % ex)
+        return 'Error decoding', 400
+
+    try:
+        r, message = read_host_config(SESSION, config)
+    except Exception as ex:
+        logging.error('Unable to parse input data: %s' % ex)
+        return 'Error parsing', 500
```

Comment:
Would there be any expected Exceptions here too that we could catch and provide a nicer error message?